extensions/blocks/contact-info/contact-info.php

```diff
@@ -10,31 +10,134 @@
  *
- * @return string
  */
-function jetpack_contact_info_block_load_assets( $attr, $content ) {
-	Jetpack_Gutenberg::load_assets_as_required( 'contact-info' );
-	return $content;
+class Jetpack_Contact_Info_Block {
+
+	/**
+	 * Adds contact info schema attributes
+	 *
+	 * @param array  $attr    Array containing the contact info block attributes.
+	 * @param string $content String containing the contact info block content.
+	 *
+	 * @return string
+	 */
+	static function render( $attr, $content ) {
+		Jetpack_Gutenberg::load_assets_as_required( 'contact-info' );
+		if( $content )
+		return str_replace(
+			'class="wp-block-jetpack-contact-info"',
+			'class="wp-block-jetpack-contact-info" itemprop="location" itemscope itemtype="http://schema.org/Organization"',
+			$content );
+	}
+
+	/**
+	 * Adds address schema attributes
+	 *
+	 * @param array  $attr    Array containing the address block attributes.
+	 * @param string $content String containing the address block content.
+	 *
+	 * @return string
+	 */
+	static function render_address( $attr, $content ) {
+		if ( ! self::has_attributes( $attr, array( 'linkToGoogleMaps' ) ) ) {
+			return ''; // nothing to see here
+		}
+		$find = array(
+			'class="wp-block-jetpack-address"',
+			'class="jetpack-address__address',// Closing " left out on purpose
+			'class="jetpack-address__region"',
+			'class="jetpack-address__city"',
+			'class="jetpack-address__postal"',
+			'class="jetpack-address__country"',
+		);
+		$replace = array(
+			'itemprop="address" itemscope itemtype="http://schema.org/PostalAddress" class="wp-block-jetpack-address" ',
+			'itemprop="streetAddress" class="jetpack-address__address', // Closing " left out on purpose
+			'itemprop="addressRegion" class="jetpack-address__region"',
+			'itemprop="addressLocality" class="jetpack-address__city"',
+			'itemprop="postalCode" class="jetpack-address__postal"',
+			'itemprop="addressCountry" class="jetpack-address__country"',
+		);
+		return str_replace( $find, $replace, $content );
+	}
+
+	/**
+	 * Adds email schema attributes
+	 *
+	 * @param array  $attr    Array containing the email block attributes.
+	 * @param string $content String containing the email block content.
+	 *
+	 * @return string
+	 */
+	static function render_email( $attr, $content ) {
+		if ( ! self::has_attributes( $attr ) ) {
+			return ''; // nothing to see here
+		}
+		return str_replace( 'href="mailto:', 'itemprop="email" href="mailto:', $content );
+	}
+
+	/**
+	 * Adds phone schema attributes
+	 *
+	 * @param array  $attr    Array containing the phone block attributes.
+	 * @param string $content String containing the phone block content.
+	 *
+	 * @return string
+	 */
+	static function render_phone( $attr, $content ) {
+		if ( ! self::has_attributes( $attr ) ) {
+			return ''; // nothing to see here
+		}
+		return str_replace( 'href="tel:', 'itemprop="telephone" href="tel:', $content );
+	}
+
+	/**
```

Comment:
I wonder if this could be useful as a helper function available to all the blocks?

Reply:
I think it would be helpful but since it is not part of any other block right now. I think it would make sense to just be part of this class.